test/replication-luatest/gh_7086_box_issue_promote_assert_test.lua

```diff
@@ -0,0 +1,233 @@
+        end, {countdown})
+    end
+end
+
+local function wal_delay_end(server)
+    server:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+end
+
+local function promote_start(server)
+    return server:exec(function()
+        local f = require('fiber').new(box.ctl.promote)
+        f:set_joinable(true)
+        return f:id()
+    end)
+end
+
+local function demote_start(server)
+    return server:exec(function()
+        local f = require('fiber').new(box.ctl.demote)
+        f:set_joinable(true)
+        return f:id()
+    end)
+end
+
+local function fiber_join(server, fiber)
+    return server:exec(function(fiber)
+        return require('fiber').find(fiber):join()
+    end, {fiber})
+end
+
+local function promote(server)
+    server:exec(function()
+        box.ctl.promote()
+        box.ctl.wait_rw()
+    end)
+end
+
+g.test_new_volatile_term_in_box_issue_promote = function(g)
+    wal_delay_start(g.server_1)
+    local f1 = promote_start(g.server_1)
+
+    wal_delay_start(g.server_2, 0)
+    local f2 = promote_start(g.server_2)
+    play_wal_until_synchro_queue_is_busy(g.server_2)
+    g.server_3:wait_election_term(g.server_2:election_term())
+
+    wal_delay_start(g.server_3, 0)
+    local f3 = promote_start(g.server_3)
+    play_wal_until_synchro_queue_is_busy(g.server_3)
+    g.server_2:wait_election_term(g.server_3:election_term())
+
+    wal_delay_end(g.server_1)
+    local ok, err = fiber_join(g.server_1, f1)
+
+    wal_delay_end(g.server_2)
+    wal_delay_end(g.server_3)
+    fiber_join(g.server_2, f2)
+    fiber_join(g.server_3, f3)
+
+    luatest.assert(not ok and err.code == box.error.INTERFERING_ELECTIONS,
+        'interfering promote not handled')
+
+    -- This leaves cluster in split brain state, lets reinit it.
+    g.cluster:drop()
+    cluster_init(g)
+end
+
+g.test_new_volatile_term_in_box_issue_demote = function(g)
+    promote(g.server_1)
+    wait_sync(g.cluster.servers)
+
+    wal_delay_start(g.server_1)
+    local f1 = demote_start(g.server_1)
+
+    wal_delay_start(g.server_2, 0)
+    local f2 = promote_start(g.server_2)
+    play_wal_until_synchro_queue_is_busy(g.server_2)
+    g.server_3:wait_election_term(g.server_2:election_term())
+
+    wal_delay_start(g.server_3, 0)
+    local f3 = promote_start(g.server_3)
+    play_wal_until_synchro_queue_is_busy(g.server_3)
+    g.server_2:wait_election_term(g.server_3:election_term())
+
+    wal_delay_end(g.server_1)
+    local ok, err = fiber_join(g.server_1, f1)
+
+    wal_delay_end(g.server_2)
+    wal_delay_end(g.server_3)
+    fiber_join(g.server_2, f2)
+    fiber_join(g.server_3, f3)
+
+    luatest.assert(not ok and err.code == box.error.INTERFERING_ELECTIONS,
+        'interfering promote not handled')
+
+    -- This leaves cluster in split brain state, lets reinit it.
+    g.cluster:drop()
+    cluster_init(g)
+end
+
+-- Demoting should fail if it is interrupted from another server
```

Comment:
`Demoting` -> `Promoting`. Demote is the next test case.

Reply:
Done.